imsim/readout.py

```diff
@@ -264,7 +262,23 @@ def finalize(self, config, base, main_data, logger):
         y_seg_offset = (0, 0, 0, 0, 0, 0, 0, 0, 2, 2, 2, 2, 2, 2, 2, 2)
         wcs = main_data[0].wcs
         crpix1, crpix2 = wcs.crpix
-        hdus = fits.HDUList(get_primary_hdu(opsim_md, det_name))
+
+        # If we don't have an OpsimMeta, then skip some header items.
+        # E.g. when reading out flat field images, most of these don't apply.
+        # The only exception is filter, which we look for in config and use that if present.
+        try:
+            opsim_md = galsim.config.GetInputObj('opsim_meta_dict', config, base,
+                                                 'CameraReadout')
+        except galsim.GalSimConfigError:
+            filt = config.get('filter', 'N/A')
```

Comment:
Should this line be something like
```
filt = base['image']['wcs'].get('band', 'N/A')
```
instead?  The current code sets `filt` to `'@image.wcs.band'`.   This is the last change needed for the ingest to work.

Reply:
No.  That was me being sloppy.  Sorry!  It should use ParseValue, rather than get.  Fixed I think.